tests/test_errors.py

```diff
@@ -1,97 +1,215 @@
-def test_in_casting_nan_error():
+def test_in_casting_nan_error() -> None:
     with pytest.raises(errors.IntCastingNaNError):
         raise errors.IntCastingNaNError
 
 
-def test_invalid_index_error():
+def test_invalid_index_error() -> None:
     with pytest.raises(errors.InvalidIndexError):
         raise errors.InvalidIndexError
 
 
-def test_merge_error():
+def test_merge_error() -> None:
     with pytest.raises(errors.MergeError):
         raise errors.MergeError
 
 
-def test_null_frequency_error():
+def test_null_frequency_error() -> None:
     with pytest.raises(errors.NullFrequencyError):
         raise errors.NullFrequencyError
 
 
-def test_numba_util_error():
+def test_numba_util_error() -> None:
     with pytest.raises(errors.NumbaUtilError):
         raise errors.NumbaUtilError
 
 
-def test_option_error():
+def test_option_error() -> None:
     with pytest.raises(errors.OptionError):
         raise errors.OptionError()
 
 
-def test_out_of_bounds_datetime():
+def test_out_of_bounds_datetime() -> None:
     with pytest.raises(errors.OutOfBoundsDatetime):
         raise errors.OutOfBoundsDatetime()
 
 
-def test_out_of_bounds_timedelta():
+def test_out_of_bounds_timedelta() -> None:
     with pytest.raises(errors.OutOfBoundsTimedelta):
         raise errors.OutOfBoundsTimedelta()
 
 
-def test_parser_error():
+def test_parser_error() -> None:
     with pytest.raises(errors.ParserError):
         raise errors.ParserError()
 
 
-def test_parser_warning():
+def test_parser_warning() -> None:
     with pytest.warns(errors.ParserWarning):
         warnings.warn("", errors.ParserWarning)
 
 
-def test_performance_warning():
+def test_performance_warning() -> None:
     with pytest.warns(errors.PerformanceWarning):
         warnings.warn("", errors.PerformanceWarning)
 
 
-def test_unsorted_index_error():
+def test_unsorted_index_error() -> None:
     with pytest.raises(errors.UnsortedIndexError):
         raise errors.UnsortedIndexError()
 
 
-def test_unsupported_function_call():
+def test_unsupported_function_call() -> None:
     with pytest.raises(errors.UnsupportedFunctionCall):
         raise errors.UnsupportedFunctionCall()
+
+
+@pytest.mark.skipif(PD_LT_15, reason="Feature added in 1.5.0")
+def test_data_error() -> None:
+    with pytest.raises(errors.DataError):
+        raise errors.DataError()
+
+
+@pytest.mark.skipif(PD_LT_15, reason="Feature added in 1.5.0")
+def test_specification_error() -> None:
+    with pytest.raises(errors.SpecificationError):
+        raise errors.SpecificationError()
+
+
+@pytest.mark.skipif(PD_LT_15, reason="Feature added in 1.5.0")
+def test_setting_with_copy_error() -> None:
+    with pytest.raises(errors.SettingWithCopyError):
+        raise errors.SettingWithCopyError()
+
+
+@pytest.mark.skipif(PD_LT_15, reason="Feature added in 1.5.0")
+def test_setting_with_copy_warning() -> None:
+    with pytest.warns(errors.SettingWithCopyWarning):
+        warnings.warn("", errors.SettingWithCopyWarning)
```

Comment:
Not sure that the reason here is correct.  `SettingWithCopyWarning` exists in 1.4, but it is in `core/common.py` and not `errors`

Part of me wants this to be right since we are still with 1.4, but the other part says to let it go because this is probably not something people end up type checking.

So can you research these and see if the locations changed in 1.4 and then update the reasons accordingly?

Reply:
I fixed these and put in a lot of comments that should be 1.5.x easier.